lib/phoenix/debug.ex

```diff
@@ -0,0 +1,88 @@
+defmodule Phoenix.Debug do
+  @moduledoc """
+  Functions for runtime introspection and debugging of Phoenix applications.
+
+  TODO
+  """
+
+  @doc """
+  Returns a list of all currently connected `Phoenix.Socket` transport processes.
+
+  Note that custom sockets implementing the `Phoenix.Socket.Transport` behaviour
+  are not listed.
+
+  Each process corresponds to one connection that can have multiple channels.
+
+  For example, when using Phoenix LiveView, the browser establishes a socket
+  connection when initially navigating to the page, and each live navigation
+  retains the same socket connection. Nested LiveViews also share the same
+  connection, each being a different channel. See `Phoenix.Debug.channels/1`.
+
+  ## Examples
+
+      iex> Phoenix.Debug.list_channel_sockets()
+      [#PID<0.123.0>]
+
+  """
+  def list_channel_sockets do
+    for pid <- Process.list(), channel_socket?(pid) do
+      pid
+    end
+  end
+
+  @doc """
+  Returns true if the given pid is a `Phoenix.Socket` transport process.
+
+  ## Examples
+
+      iex> Phoenix.Debug.list_channel_sockets() |> Enum.at(0) |> channel_socket?()
+      true
+
+      iex> channel_socket?(pid(0,456,0))
+      false
+
+  """
+  def channel_socket?(pid) do
```

Comment:
Perhaps it should be private?

Reply:
I think it's fine to have it public.

Reply:
I am just bothered about the name a bit. We have `socket` which returns a `Phoenix.Socket` and this is not it. :( I guess the same issue applies to `list_channel_sockets` though...

Reply:
Maybe `channel_socket_process?` and `list_channel_socket_info` / `list_channel_sockets_info` 🤷🏻‍♂️

Reply:
alright, private it is!

Reply:
oh fwiw in the LiveView PR I called it `liveview_process?`: https://github.com/phoenixframework/phoenix_live_view/pull/3776/files#diff-073229604d21f1e1dbd1c71bbd2448931f7a0a9bfc99fcbe3fb2e7a3882e1049R89

Reply:
That works then!

Reply:
So should we add it as `channel_socket_process?`?

But now I'm wondering if we should just go with `list_sockets` and `socket_process?` and just clearly state that it only lists sockets defined with `use Phoenix.Socket` and not custom `Phoenix.Socket.Transport`s?

Reply:
list_socket and socket_process? Works for me.

Reply:
Alright, we're back at [this comment](https://github.com/phoenixframework/phoenix/pull/6223#discussion_r2073083243) then, but I think it's fine with an explanation in the docs.